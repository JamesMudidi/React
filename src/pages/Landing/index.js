import React from 'react';
import { Jumbotron } from 'react-bootstrap';
import Head from '../../components/Navbar';
import Body from '../../components/Body';
import Footer from '../../components/Footer';
import Incident from '../../components/Incident';

const username = sessionStorage.getItem('username');
export const isLoggedInUser = () => {
  let placeholder = 'Logout';
  if (username === null) {
    placeholder = 'Sign up';
  }
  return placeholder;
};
export const isLoggedOutInUser = () => {
  let placeholder = username;
  if (username === null) {
    placeholder = 'Login';
  }
  return placeholder;
};
export const isAuthenticated = () => {
  let placeholder = false;
  if (username === null) {
    placeholder = true;
  }
  return placeholder;
};

const Landing = () => (
  <div>
    <Head loginLink={isLoggedOutInUser()} signupLink={isLoggedInUser()} />
    {isAuthenticated() && <Body />}
    {!isAuthenticated() && <div><Jumbotron><Incident /></Jumbotron></div>}
    <Footer />
  </div>
);

export default Landing;
