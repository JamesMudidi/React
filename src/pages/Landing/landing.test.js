import React from 'react';
import { shallow } from 'enzyme';
import Landing from 'pages/Landing';

// test to check if the home component matches the snapshot
describe('Login Component', () => {
  it('matches the snapshot', () => {
    const component = shallow(<Landing />);
    expect(component).toMatchSnapshot();
  });
});
