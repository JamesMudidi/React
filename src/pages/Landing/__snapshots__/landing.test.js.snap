// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component matches the snapshot 1`] = `
<div>
  <Head />
  <Body />
  <Footer />
</div>
`;
